source/js/fields.js

```diff
@@ -334,48 +334,74 @@ class Fields {
             const checkboxGroups = form.querySelectorAll('.checkbox-group-required');
-
-
+            
             inputs.forEach(input => {
                 if(input.hasAttribute('data-validation-message')) {
                     this.getFieldWrapper(input).querySelector('.c-field__error').setAttribute('aria-label', input.getAttribute('data-validation-message'));
                     this.getFieldWrapper(input).querySelector('.c-field__error-message').innerHTML = input.getAttribute('data-validation-message');
 
-
-                    input.addEventListener('keyup', () => {
-                        if (this.getFieldWrapper(input).classList.contains('is-invalid') || this.getFieldWrapper(input).classList.contains('is-valid')) {
-                            self.validateInput(input);
-                        }
-                    })
                 } else {
                     if (this.getFieldWrapper(input).querySelector('.c-field__error')) {
                         this.getFieldWrapper(input).querySelector('.c-field__error').remove();
                     }
                 }
+
+                input.addEventListener('keyup', () => {
+                    if (this.getFieldWrapper(input).classList.contains('is-invalid') || this.getFieldWrapper(input).classList.contains('is-valid')) {
+                        self.validateInput(input);
+                    }
+                })
+            });
+
+            form.querySelectorAll('.checkbox-group-required').forEach(checkboxGroup => {
+                const checkboxes = checkboxGroup.querySelectorAll('.c-option__checkbox--hidden-box');
+                let validationElement = checkboxGroup.querySelector('.js-checkbox-valid');
+
+                checkboxes.forEach(checkbox => {
+                    checkbox.addEventListener('change', () => {
+                        let validator = checkboxGroup.querySelectorAll('.c-option [type="checkbox"]:checked');
+                        if (validator.length > 0) {
+                            validationElement.setAttribute('checked', true);
+                            checkboxGroup.querySelector('.c-field__label').classList.remove('u-color__text--danger');
+                        } else {
+                            validationElement.removeAttribute('checked');
+                        }
+                    })
+                })
             });
 
             // Validate fields on change
             ['focusout', 'change'].forEach(function(e) {
                 inputs.forEach(input => {
-                    input.addEventListener(e, () => self.validateInput(input));
+                    input.addEventListener(e, () => {
+                        self.validateInput(input) 
+                    });
                 });
             });
 
-            //General validation errors
-            form.addEventListener('invalid', (e) => {
+            submitButton.addEventListener('click', (e) => {
```

Comment:
Do not remove the invalid event listerna. This enshures that all submit events are covered.